### ISSUE TITLE
JSON-RPC errors are sent as string instead of object, not following JSON-RPC spec

### ISSUE THREAD

**Issue by SomberNight** [romanz/electrs#313]
Protocol errors are not sent correctly to the client. See https://www.jsonrpc.org/specification#error_object

-----

Example:
when I connect to Electrs and subscribe to a large history address, I get this message:
`{'error': 'Too many history entries', 'id': 11, 'jsonrpc': '2.0'}`
Electrum (the client) will then log:
```
I/i | interface.[cryptobadger.io:50002] | connection established. version: ['electrs 0.4.1', '1.4']
[...]
E/i | interface.[cryptobadger.io:50002] | Exception in wrapper_func: ProtocolError(-32600, 'ill-formed response error object: Too many history entries')
Traceback (most recent call last):
  File "...\electrum\electrum\util.py", line 1039, in wrapper
    return await func(*args, **kwargs)
  File "...\electrum\electrum\interface.py", line 480, in wrapper_func
    return await func(self, *args, **kwargs)
  File "...\electrum\electrum\interface.py", line 504, in run
    await self.open_session(ssl_context)
  File "...\electrum\electrum\interface.py", line 644, in open_session
    await group.spawn(self.monitor_connection)
  File "...\aiorpcX\aiorpcx\curio.py", line 242, in __aexit__
    await self.join()
  File "...\aiorpcX\aiorpcx\curio.py", line 211, in join
    raise task.exception()
  File "...\electrum\electrum\synchronizer.py", line 81, in _start_tasks
    await group.spawn(self.main())
  File "...\aiorpcX\aiorpcx\curio.py", line 242, in __aexit__
    await self.join()
  File "...\aiorpcX\aiorpcx\curio.py", line 211, in join
    raise task.exception()
  File "...\electrum\electrum\synchronizer.py", line 109, in subscribe_to_address
    await self.session.subscribe('blockchain.scripthash.subscribe', [h], self.status_queue)
  File "...\electrum\electrum\interface.py", line 194, in subscribe
    result = await self.send_request(method, params)
  File "...\electrum\electrum\interface.py", line 170, in send_request
    response = await asyncio.wait_for(
  File "...\Python38\lib\asyncio\tasks.py", line 455, in wait_for
    return await fut
  File "...\aiorpcX\aiorpcx\session.py", line 537, in send_request
    return await self._send_concurrent(message, future, 1)
  File "...\aiorpcX\aiorpcx\session.py", line 507, in _send_concurrent
    return await future
  File "...\aiorpcX\aiorpcx\jsonrpc.py", line 721, in receive_message
    item, request_id = self._protocol.message_to_item(message)
  File "...\aiorpcX\aiorpcx\jsonrpc.py", line 273, in message_to_item
    return cls._process_response(payload)
  File "...\aiorpcX\aiorpcx\jsonrpc.py", line 220, in _process_response
    raise cls._error(code, message, False, request_id)
aiorpcx.jsonrpc.ProtocolError: (-32600, 'ill-formed response error object: Too many history entries')
```

If I do the same but connect to ElectrumX, I get:
`{'jsonrpc': '2.0', 'error': {'code': 1, 'message': 'history too large'}, 'id': 11}`
Electrum (the client) will then log:
```
I/i | interface.[fortress.qtornado.com:443] | connection established. version: ['ElectrumX 1.15.0', '1.4']
[...]
E/i | interface.[fortress.qtornado.com:443] | disconnecting due to GracefulDisconnect(RPCError(1, 'history too large'))
```

**Comment by romanz** [comment]
Thanks for reporting this issue!

**Comment by romanz** [comment]
I think it should be resolved in `next` - in case of an error it sends:
```
{"error":{"code":1,"message":"..."},"id":11,"jsonrpc":"2.0"}
```

**Comment by knocte** [comment]
> I think it should be resolved in next branch - in case of an error it sends:

Hey @romanz what do you mean with `next` branch? Given that this issue had been closed, I was assuming that it's fixed, but even the last version of electrs (0.8.9) doesn't seem to contain a fix for this. (So then this shouldn't have been closed until that branch is merged?)

**Comment by Kixunil** [comment]
@knocte `next` branch is a new experimental branch. I think the expectation was that it'd become master soon but we hit some issues.

I agree that closing this only after the fix is in `master` would be better.

**Comment by romanz** [comment]
Thanks for the heads up!

**Comment by romanz** [comment]
Should be fixed by #390.
